src/test_creation/modules/code_analyzer/repo.py

```diff
@@ -44,7 +45,19 @@ def _get_language_file_map(self):
             for k, v in self.fileext_language_map.items():
                 if file.endswith(k):
                     file_language_map[v].append(file)
-        return file_language_map
+        return file_language_map # FIXME: why is it called file_language_map instead of language_file_map?
+
+    def _get_file_function_lineno_map(self):
+        file_function_lineno_map = defaultdict(lambda: defaultdict(int))
+        files = self.lf_map.get("Python", [])
+        ast = PythonASTCodeAnalyzer() # FIXME: only support Python ATS, what's the implication?
```

Comment:
@SoloSynth1, for now I only get the line numbers using ATS, what's the implication if ATS is failed?

Reply:
The `AST` module only supports [grammar of the current version of the Python you're running](https://docs.python.org/3/library/ast.html#:~:text=The%20abstract%20syntax%20itself%20might%20change%20with%20each%20Python%20release%3B%20this%20module%20helps%20to%20find%20out%20programmatically%20what%20the%20current%20grammar%20looks%20like.).

If AST fails to parse the code, there could be three possibilities:
1. The content is grammatically incorrect;
2. The code was written using older version of Python (e.g. Python 2). Grammatically speaking, there were little breaking changes across minor version bumps, so all Python 3 codes should be parse without problems; or
3. The code was written with newer syntaxes introduced in the future versions, assuming the user keeps using the same version as we do right now i.e. 3.12

So, to answer your question, if AST returns errors, there could be a chance (a very likely one) that the file contains valid Python codes, but they are not 100% conform to the current Python version that we are running the system. In fact during the initial analysis of the ~300 repos I discovered that a considerable amount of them contain Python 2 codes, and as I have written a naive code parser as a fallback.

Reply:
Noted, thanks. I have created #123 for this.